lowranceusr.cc

```diff
@@ -758,24 +1802,140 @@ lowranceusr_waypt_disp(const Waypoint* wpt)
   /* USER waypoint type */
   short int WayptType = 0;
   gbfputint16(WayptType, file_out);
+
+  if (writing_version == 3) {
+    float depth = WAYPT_HAS(wpt, depth) ?
+                  METERS_TO_FEET(wpt->depth) : -99999.0;
+    gbfputint32(depth, file_out);
```

Comment:
L1808 implies the magic value -99999.0 indicates no data.
L1809 converts the float depth to an int, and saves it as an int, not a float.
Compare to comment at L973-974.
These operations should be symmetric between read and write.